srtcore/threadname.h

```diff
@@ -68,8 +76,25 @@ class ThreadName
 class ThreadName
 {
 public:
+    static const size_t BUFSIZE = 128;
 
-    static bool get(char*) { return false; }
+    static bool get(char* output)
+    {
+        // The default implementation will simply try to get the thread ID
+#if USE_STDCXX_SYNC
+        std::ostringstream bs;
+        bs << "T" << std::this_thread::get_id();
+        size_t s = bs.str().copy(output, BUFSIZE-1);
+        output[s] = '\0';
+        return true;
+#elif _WIN32
+        sprintf(output, "T%uX", GetCurrentThreadId());
+        return true;
+#else
+        sprintf(output, "T%luX", (unsigned long)pthread_self());
+        return true;
+#endif
```

Comment:
What if make use of `srt::sync::this_thread::get_id()` instead of:
- `std::this_thread::get_id()`
` `pthread_self()`